velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -3109,3 +3109,169 @@ TEST_F(DateTimeFunctionsTest, timeZoneMinute) {
       timezone_minute("2023-", "Pacific/Chatham"),
       "Unable to parse timestamp value: \"2023-\", expected format is (YYYY-MM-DD HH:MM:SS[.MS])");
 }
+
+TEST_F(DateTimeFunctionsTest, toISO8601TestVarchar) {
+  const auto toISO8601 = [&](std::optional<Date> dateObj) {
+    return evaluateOnce<std::string, Date>("to_iso8601(c0)", dateObj);
+  };
+
+  // Date() input: days since epoch
+
+  // Date(0) is 1970-01-01.
+  EXPECT_EQ("1970-01-01T00:00:00.000", toISO8601(Date()));
+
+  // Date(18297) is 2020-02-05.
+  EXPECT_EQ("2020-02-05T00:00:00.000", toISO8601(Date(18297)));
+
+  // Date(-18297) is 1919-11-28.
+  EXPECT_EQ("1919-11-28T00:00:00.000", toISO8601(Date(-18297)));
+
+  // Date(980127) is 4653-07-01.
+  EXPECT_EQ("4653-07-01T00:00:00.000", toISO8601(Date(980127)));
+
+  // Date(-45734) is 1844-10-14.
+  EXPECT_EQ("1844-10-14T00:00:00.000", toISO8601(Date(-45734)));
+}
+
+TEST_F(DateTimeFunctionsTest, toISO8601TestTimestamp) {
```

Comment:
Similar comment as Date.... Its better to take a string representation of the timestamp, parse it to a timestamp and then call toISO8601 on it for a cleaner test.

Reply:
Done.